### ISSUE TITLE
Key change from EC256 to RSA4096 failed

### ISSUE THREAD

**Issue by notasecret** [peacemakr-io/peacemakr-go-sdk#50]
For example,

```
[jonmclachlan@Jons-MacBook-Pro-2 .peacemakr]$ ls
io.peacemakr.asymmetric.np-zt68uq3l2ypzmfkhfwmdf1tfd31po9vs1295wp-y-	io.peacemakr.pub
io.peacemakr.clientid							io.peacemakr.symmetric.5pc8pmkzwlvtej46ayqvrcfgty1iwfisnjbaqgrelqu-
io.peacemakr.key-creation-time						io.peacemakr.symmetric.douxvyw-qiuxko-o8kksd-pvjojyb22r8kw3iiagpqo-
io.peacemakr.keyid							io.peacemakr.symmetric.ub9x4vn00w8mhwgbcihejtijs1rhrbgbije-t4rxav8-
io.peacemakr.priv
[jonmclachlan@Jons-MacBook-Pro-2 .peacemakr]$ cat io.peacemakr.pub
-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEKKnzMeX7neIIz3ugvKq5V4EZeDf/
Z8SQonwF+tuNdXP5+pm3D1ZvBCPNwCJ3g5Fu2pRM5zcSZOwWDvleH2DJYA==
-----END PUBLIC KEY-----
[jonmclachlan@Jons-MacBook-Pro-2 .peacemakr]$ 
[jonmclachlan@Jons-MacBook-Pro-2 .peacemakr]$ 
[jonmclachlan@Jons-MacBook-Pro-2 .peacemakr]$ 
[jonmclachlan@Jons-MacBook-Pro-2 .peacemakr]$ echo "secret stuff" | time peacemakr-cli -action encrypt
MyProjectCrypto2020/12/08 19:07:44 [/Users/dahuang/go/pkg/mod/github.com/peacemakr-io/peacemakr-go-sdk@v0.0.11-0.20201019035216-2b49761dc8f0/pkg/crypto_impl.go: 1339] ClientDebugInfo *** clientId = my-client-id, clientName = peacemakr-cli, org id = PeacemakrTestOrgId-89A490E7-ED4D-497B-963B-F200DA3E8F96, version = 0.0.1 : Rotating key because asymmetric key config changed from 3 to 2
MyProjectCrypto2020/12/08 19:07:50 [/Users/dahuang/go/pkg/mod/github.com/peacemakr-io/peacemakr-go-sdk@v0.0.11-0.20201019035216-2b49761dc8f0/pkg/crypto_impl.go: 1408] ClientDebugInfo *** clientId = my-client-id, clientName = peacemakr-cli, org id = PeacemakrTestOrgId-89A490E7-ED4D-497B-963B-F200DA3E8F96, version = 0.0.1 : Error on the network, rolling back public key changes
MyProjectCrypto2020/12/08 19:07:50 [/Users/dahuang/go/pkg/mod/github.com/peacemakr-io/peacemakr-go-sdk@v0.0.11-0.20201019035216-2b49761dc8f0/pkg/crypto_impl.go: 1409] ClientDebugInfo *** clientId = my-client-id, clientName = peacemakr-cli, org id = PeacemakrTestOrgId-89A490E7-ED4D-497B-963B-F200DA3E8F96, version = 0.0.1 : context deadline exceeded
AQAAAAkAAAAoAAAALAAAADAAAABEAAAAWAAAALsAAADMAAAAGAEAAAAEAAIAAAAAEAAAAAjx
Q3m2UbosaJm9Uz4g6q4QAAAAXH7uIP88agBv5+zisuLMFF8AAAB7ImNyeXB0b0tleUlEIjoi
ZE91eFZ5dytRSXV4a28rTzhra3NkL3BWam9qeWIyMlI4a3czSWlBZ3Bxbz0iLCJzZW5kZXJL
ZXlJRCI6Im15LXB1YmxpYy1rZXktaWQifQ0AAAAgjK4orYb8AujeUdutSAAAADBGAiEAuAYO
PQQgESU3KwaOQjYQx/DlHbsmj5Mu9xLH6zg5XS0CIQCjzmj34BE9SCtYewZ9BIRlXo1Du1Cd
HHTknKcBV+yaBLHJtBhI/WXHQqAVRmgsOgLja49OSrB2y3nPVREKpjgC
        6.61 real         4.72 user         0.21 sys
```

Even after repeated attempts, same behavior

**Comment by notasecret** [comment]
(also happens on the other direction, too, from RSA4096 -> EC256)

```
[jonmclachlan@Jons-MacBook-Pro-2 tmp]$ echo "secret stuff" | time peacemakr-cli -action encrypt
MyProjectCrypto2020/12/08 19:13:56 [/Users/dahuang/go/pkg/mod/github.com/peacemakr-io/peacemakr-go-sdk@v0.0.11-0.20201019035216-2b49761dc8f0/pkg/crypto_impl.go: 1339] ClientDebugInfo *** clientId = Sm96/Vfhy9VynsUE7ZJ3CgL5GUR5ZdK9LNRLtYANTuk=, clientName = peacemakr-cli, org id = PeacemakrTestOrgId-89A490E7-ED4D-497B-963B-F200DA3E8F96, version = 0.0.1 : Rotating key because asymmetric key config changed from 2 to 3
MyProjectCrypto2020/12/08 19:13:56 [/Users/dahuang/go/pkg/mod/github.com/peacemakr-io/peacemakr-go-sdk@v0.0.11-0.20201019035216-2b49761dc8f0/pkg/crypto_impl.go: 1408] ClientDebugInfo *** clientId = Sm96/Vfhy9VynsUE7ZJ3CgL5GUR5ZdK9LNRLtYANTuk=, clientName = peacemakr-cli, org id = PeacemakrTestOrgId-89A490E7-ED4D-497B-963B-F200DA3E8F96, version = 0.0.1 : Error on the network, rolling back public key changes
MyProjectCrypto2020/12/08 19:13:56 [/Users/dahuang/go/pkg/mod/github.com/peacemakr-io/peacemakr-go-sdk@v0.0.11-0.20201019035216-2b49761dc8f0/pkg/crypto_impl.go: 1409] ClientDebugInfo *** clientId = Sm96/Vfhy9VynsUE7ZJ3CgL5GUR5ZdK9LNRLtYANTuk=, clientName = peacemakr-cli, org id = PeacemakrTestOrgId-89A490E7-ED4D-497B-963B-F200DA3E8F96, version = 0.0.1 : context deadline exceeded
AQAAAAkAAAAoAAAALAAAADAAAABEAAAAWAAAANcAAADoAAAA7AIAAAAEAAIAAAAAEAAAAIfX
AdVou2vnIz9vyoc9ZqAQAAAAIBjSLWIZyimNkKxDv6CoynsAAAB7ImNyeXB0b0tleUlEIjoi
ZE91eFZ5dytRSXV4a28rTzhra3NkL3BWam9qeWIyMlI4a3czSWlBZ3Bxbz0iLCJzZW5kZXJL
ZXlJRCI6Im55UVFOM1ByTXRqbnpjUitkbk5ialQrSWdZd1VwVER5OXQxbG40MjJHeXM9In0N
AAAAQpiwfTvleXA8DFNHowACAACG/zJFWLaxBKbhLef/c9+5kOU+5XKQo36deWPdVnowgFXQ
FZJzlpdNRuPolm2qlooBBNzYgJLDvYv+kO44Emax4lHkvH+j7bMRz/QyDzeXmsWVfyjgGEu4
PwgB5bN66ZInZCL+t4QaKQ0RNL+qYFSABxvNI907D7hm7YsfDf2eBdHnxp9fTzF+5l/X7xIw
7/DJjm3xil5soL8mVjW/eE5d2dZu/uKq0YOQsjicHEeOjbrVFwvVr1eq8eZ9qeWKkjwlATtL
KV9FB0MCLDoWw8wFbEdCS0edo18NlIfNis/Nwi/EvoxnjR4HtT21EipznHoGJ7/YWTor8FKj
DoQQSPRnjXUXwo+vS8CzI9mr6HKBiuB5KdhoSsFcG8L5kDu6Bo8ASoY+2Y47emxh7mn8IuZd
fmGCnwJGsohp966tYbvsMQzYCV9z+VA3pRUV04CE8KObcCofQ3ppgQ/ZdHcInLBVVOpIxDL9
g8Xi+1jHQ8B2fjQRwstIi3GJHIYdkZqmX5LnUwZS2T0XqeS4pZG5hHbDfvZ2gH5drI0w0iZ1
wSOll0AcvoElLEO1GsCAvmK0qvh0WZp6snR9Q+hi7L2e/vJl+eLa9Ymy4S8uWCsGgBzQ6hNc
XM/zfZ/Cnu0+yk3ZG90qHMv8DzZTrN2udjUSfM5TSLuz8/nZQNUcaMxqlL/0LAj3SvFR9B/w
Tw7By0VPUQ6ayL6OAHWhakooloM/F8xD
        0.87 real         0.14 user         0.05 sys
[jonmclachlan@Jons-MacBook-Pro-2 tmp]$ rm -rf /tmp/.peacemakr/
[jonmclachlan@Jons-MacBook-Pro-2 tmp]$ echo "secret stuff" | time peacemakr-cli -action encrypt
AQAAAAkAAAAoAAAALAAAADAAAABEAAAAWAAAANcAAADoAAAAMwEAAAAEAAIAAAAAEAAAANi9
9YFhaoVmzmXkXX4wCBEQAAAAPAOg+qpPQAFkxVKp0mh1InsAAAB7ImNyeXB0b0tleUlEIjoi
ZE91eFZ5dytRSXV4a28rTzhra3NkL3BWam9qeWIyMlI4a3czSWlBZ3Bxbz0iLCJzZW5kZXJL
ZXlJRCI6ImVIM2xrTVZzNDlJd2lkUFlUV3ZxZmhnMGcrbFg2N3FhTUVWRXhPbHMyekk9In0N
AAAAKK3lhNyAOjtOTi0YQEcAAAAwRQIgeaFocV4yZSO0b1SFVPwYzOcjWY5Hv58hEk13c5Mv
wzkCIQCACaNPq60apcNso/nWABt658wyAw+l8FySs07kRS2lIg3XnNpPY18sfZ+XZokN5ZXu
G1QsdUX7SSpSyte62Dyv
        5.77 real         0.13 user         0.05 sys
[jonmclachlan@Jons-MacBook-Pro-2 tmp]$ 
```

**Comment by notasecret** [comment]
The change was in the admin portal.

**Comment by notasecret** [comment]
Also noticed that `/Users/dahuang/go/pkg/mod` is not my gopath, but that one is more forgivable, since it does not break the software :-p

**Comment by yhung119** [comment]
resolved by #51 
Reason: Timeout was not set when setting up parameters for adding public key.
